### PR TITLE
Add models

### PR DESCRIPTION
**Pull request by JonOttesen**
Added EfficientNet3D
Added plotting of training/validation loss when saving checkpoints
Removed unnecessary sanity check

### CODE REVIEW

File: crainets/essentials/multi_tracker.py [hunk 1]
```diff
@@ -112,6 +116,48 @@ def training_metric(self, epoch):
...
+        validation = list()
+        x = np.arange(len(train.items()))
+
+        for k, (i, j) in enumerate(train.items()):
+            training.append(np.mean(np.array(j['loss'])))
+
+        for k, (i, j) in enumerate(valid.items()):
+            validation.append(np.mean(np.array(j['loss'])))
+
+        cm = 1/2.54
```
Comment (JonNesvold) [inline crainets/essentials/multi_tracker.py:138]: I would give these vars a more explicit name and place them in config instead. It clutters the code

File: crainets/essentials/multi_tracker.py [hunk 2]
```diff
@@ -112,6 +116,48 @@ def training_metric(self, epoch):
...
+        w = 17.6
+        fig, ax = plt.subplots(figsize=(w*cm, 2*w/3*cm))
```
Comment (JonNesvold) [inline crainets/essentials/multi_tracker.py:140]: Let me know what is going on with the arithmetic in a comment please

File: crainets/models/efficientnet3d/efficientnet3d.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+        # Batch norm parameters
+        self.batch_norm_momentum = 1 - global_params.batch_norm_momentum
+        self.batch_norm_epsilon = global_params.batch_norm_epsilon
+
+        # Use dynamic padding, I see no reason not to use this
+        Conv3d = Conv3dDynamicSamePadding
+
+        # Stem
+        stem_stride = 2
```
Comment (JonNesvold) [inline crainets/models/efficientnet3d/efficientnet3d.py:78]: Can you place global vars and constants in config? It clutters :)

File: crainets/models/efficientnet3d/mbconvblock3d.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        if self.id_skip and self.stride == 1 and self.in_channels == self.out_channels:
+            # The combination of skip connection and drop connect brings about stochastic depth.
+            if drop_connect_rate:
+                x = drop_connect(x, p=drop_connect_rate, training=self.training)
+            x = x + inputs  # skip connection
+
+        return x
+
+
+    def _norm(self, norm: str, output: int, bias: bool):
```
Comment (JonNesvold) [inline crainets/models/efficientnet3d/mbconvblock3d.py:147]: This function could be made nicer with dicts and setattr. Repeated code is a pain to maintain

**Review by JonNesvold — APPROVED** [review]
Nice work! Super nice that you have added on many new features! The one thing that bugs me a bit with the code is that there is a lot going on in every single script. What I mean by that is: too many variable declarations, repetitive patterns and magic numbers.

You are free to decide whether or not you'll do something about because it's not bad but I personally think that it's room for some improvements :)
